jaeger_client/sampler.py

```diff
@@ -421,56 +421,64 @@ def _sampling_request_callback(self, future):
 
         response = future.result()
         try:
-            sampler, strategies = \
-                self._parse_sampling_strategy(self.sampler, response.body)
+            sampling_strategies_response = json.loads(response.body)
         except Exception as e:
             self.error_reporter.error(
                 Metrics.SAMPLER_ERRORS, 1,
                 'Fail to parse sampling strategy '
                 'from jaeger-agent: %s [%s]', e, response.body)
             return
 
-        with self.lock:
-            if isinstance(sampler, AdaptiveSampler):
-                sampler.update(strategies)
-            elif self.sampler == sampler:
-                return
-            self.sampler = sampler
-        self.logger.debug('Tracing sampler set to %s', sampler)
-
-    def _poll_sampling_manager(self):
-        self.logger.debug('Requesting tracing sampler refresh')
-        fut = self._channel.request_sampling_strategy(
-            self.service_name, timeout=15)
-        fut.add_done_callback(self._sampling_request_callback)
+        self._update_sampler(sampling_strategies_response)
+        self.logger.debug('Tracing sampler set to %s', self.sampler)
 
-    def close(self):
+    def _update_sampler(self, response):
         with self.lock:
-            self.running = False
-            if self.periodic is not None:
-                self.periodic.stop()
+            if response.get(OPERATION_SAMPLING_STR):
+                self._update_adaptive_sampler(response.get(OPERATION_SAMPLING_STR))
+            else:
+                try:
```

Comment:
it doesn't know, but might in the future. Instead of catching exceptions at multiple levels I would rather catch them once in _update_sampler, to ensure that this thread doesn't die.